### DISCUSSION TITLE
Please,where's train unet only?

### DISCUSSION THREAD

**Discussion by tsjslgy** [bmaltais/kohya_ss#1277] [Q&A] [answered]
From kohya_ss 12GB GPU SDXL training:  Use --network_train_unet_only.
koyha_ss_gui. Where's set?  Thanks

**Comment by danielpdykes** [comment, marked as answer]
@tsjslgy Under 'Training', click 'Advanced Configuration' 
Search for 'Additional parameters'
Enter "--network_train_unet_only" into the input field

**Comment by avan06** [comment]
Currently, "network_train_unet_only" seems to be automatically determined whether to include it or not. In the following code snippet from lora_gui.py, when "text_encoder_lr" is 0 and "unet_lr" is not 0, it will be automatically added.

```py
if not (float(text_encoder_lr) == 0) or not (float(unet_lr) == 0):
    if not (float(text_encoder_lr) == 0) and not (float(unet_lr) == 0):
        run_cmd += f' --text_encoder_lr={text_encoder_lr}'
        run_cmd += f' --unet_lr={unet_lr}'
    elif not (float(text_encoder_lr) == 0):
        run_cmd += f' --text_encoder_lr={text_encoder_lr}'
        run_cmd += f' --network_train_text_encoder_only'
    else:
        run_cmd += f' --unet_lr={unet_lr}'
        run_cmd += f' --network_train_unet_only'
```

The author of sd-scripts, kohya-ss, provides the following recommendations for training SDXL:

> kohya-ss:
> Please specify `--network_train_unet_only` if you caching the text encoder outputs.
> 
> For the second command, if you don't use the option `--cache_text_encoder_outputs`, Text Encoders are on VRAM, and it uses a lot of VRAM. So please add the option (and also add `--network_train_unet_only`).
> https://github.com/kohya-ss/sd-scripts/issues/661#issuecomment-1643945086

I wonder if the developer of kohya_ss_gui has considered adding an option in the GUI interface for users to decide whether to enable "network_train_unet_only"?

**Comment by araleza** [comment]
I think the Guides tab in kohya_ss might be wrong.  It says:

> We strongly recommend using the --train_unet_only option for SDXL LoRA to avoid unforeseen training results caused by dual text encoders in SDXL.

But the correct option seems to be `--network_train_unet_only`, not `--train_unet_only`.
